code/SingleRecordAdmin.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace SilverStripe\Admin;
+
+use SilverStripe\Forms\Form;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * The base class for admin sections that exist for editing a single record.
+ */
+abstract class SingleRecordAdmin extends LeftAndMain
+{
+    /**
+     * Determines if there should be a single record in the database that this admin edits.
+     *
+     * If this is not true, you need to provide a mechanism to tell the form which record it should be editing.
+     * This could be an action (e.g. edit/$ID), or could be based on something about the current member, etc.
+     */
+    private static bool $only_one_record = true;
+
+    /**
+     * If no record exists, allow a new one to be created
+     */
+    private static bool $allow_new_record = true;
+
+    public function getEditForm($id = null, $fields = null): ?Form
+    {
+        $form = parent::getEditForm($id, $fields);
+        // Keep the tabs on the top row, rather than underneath
+        if ($form?->Fields()->hasTabSet()) {
+            $form->Fields()->findOrMakeTab('Root')->setTemplate('SilverStripe/Forms/CMSTabSet');
+        }
+        return $form;
+    }
+
+    public function getRecord($id): ?DataObject
+    {
+        if (!$id) {
+            return $this->getSingleRecord();
+        }
+        return parent::getRecord($id);
```

Comment:
What's the scenario where $id is truthy?  Seems like we should be throwing an exception here instead?

Reply:
`$id` is truthy if a valid ID is passed in. So we either want the "single record" based on the configuration (where a `null` or `0` is passed in), or we want the specified record if a specific ID is passed in.

Reply:
Why would be specify an $id though in a SingleRecordAdmin? Isn't the whole point to only have a single record?